### PR TITLE
Try to enable loading of https URLs

### PR DESCRIPTION
**Pull request by koppor**
For the study, I need to have a way of quick loading a repository via https URL. This is a first try, but does not work currently.

### CODE REVIEW

File: src/plugins/api.js [hunk 1]
```diff
@@ -186,19 +186,43 @@ export async function pushToGitHub(newCommitSha) {
...
+                return "[]";
+            });
+        return repoInfo.then(
+            repoInfo => Array.of(repoInfo)
+        )
+    } else {
+        return pizzly
+            .integration("github")
+            .auth(auth)
+            .get("/auth/repos?sort=updated&page=" + page + "&per_page=" + per_page)
```
Comment (KLBa) [inline src/plugins/api.js:220]: Replace `/auth` by `/user`, i.e., this line should be `.get("/user/repos?sort=updated&page=" + page + "&per_page=" + per_page)`

File: src/plugins/api.js [hunk 2]
```diff
@@ -186,19 +186,43 @@ export async function pushToGitHub(newCommitSha) {
...
+        let url = "/repos/" + match[1] + "/" + match[2];
+        console.log("url: " + url);
+        // auth entered URL
+        let repoInfo = pizzly
+            .integration("github")
+            .auth(auth)
+            .get(url)
+            .then(response => {
+                console.log(response)
+                response.json()
```
Comment (KLBa) [inline src/plugins/api.js:207]: Add the `return` statement, i.e., `return response.json()`

**Review by KLBa — COMMENTED** [review]
The problem is mostly that you changed the url in the "old" code. As a result, there is an error right at the start.
I annotated the two lines you need to change. With those changes it worked for me:

grafik

Two notes:

* To fix the regular search functionality you should also pass an empty searchstring to the call to `loadRepositoryList` in line 248. Is there a particular reason why you changed `loadRepositoryList` and not `searchRepositoryList`?
* Note that I was trying to do something similar using the full name on my branch adr-parser-analysis. (However, my version might have caused more problems for you anyway, because the current search function would have loaded all your private repositories first.) Anyway I prefer your solution with using the URL as its more "hidden".

**Comment by koppor** [comment]
@KLBa Thank you four your comments! Now, it works 🎉

**Pull request merged**